azure-pipelines/pipelines.yml

```diff
@@ -151,10 +151,10 @@ jobs:
       displayName: "Build vcpkg with CMake"
       inputs:
         script: |
-          call "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\Tools\VsDevCmd.bat" -arch=x86 -host_arch=x86
-          rmdir /s /q build.x86.debug > nul 2> nul
-          cmake.exe -G Ninja -DCMAKE_BUILD_TYPE=Debug -DBUILD_TESTING=ON -DVCPKG_DEVELOPMENT_WARNINGS=ON -DVCPKG_WARNINGS_AS_ERRORS=ON -DVCPKG_BUILD_BENCHMARKING=ON -DVCPKG_BUILD_FUZZING=ON -DVCPKG_BUILD_TLS12_DOWNLOADER=ON -DVCPKG_ARTIFACTS_DEVELOPMENT=ON -B build.x86.debug
-          ninja.exe -C build.x86.debug all generate-message-map
+          call "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\Tools\VsDevCmd.bat" -arch=x64 -host_arch=x64
+          rmdir /s /q build.x64.debug > nul 2> nul
+          cmake.exe -G Ninja -DCMAKE_C_COMPILER=clang-cl -DCMAKE_CXX_COMPILER=clang-cl -DCMAKE_BUILD_TYPE=Debug -DBUILD_TESTING=ON -DVCPKG_DEVELOPMENT_WARNINGS=ON -DVCPKG_WARNINGS_AS_ERRORS=ON -DVCPKG_BUILD_BENCHMARKING=ON -DVCPKG_BUILD_FUZZING=ON -DVCPKG_BUILD_TLS12_DOWNLOADER=ON -DVCPKG_ARTIFACTS_DEVELOPMENT=ON -B build.x64.debug
+          ninja.exe -C build.x64.debug all generate-message-map

```

Comment:
who had the idea to call ninja directly instead of calling ninja through cmake? is there a good reason to do it this way?

Reply:
Adding more indirection didn't seem to be helpful?

Reply:
depends if you want to use presets or not. All these cmake calls look like a prime example to use cmake presets so that people can easily reproduce them locally.

Reply:
This script long predates cmake presets being a thing.

Reply:
> New in version 3.19.

```
From 222bf361e4f0f75698eab0bcad55064cec539923 Mon Sep 17 00:00:00 2001
From: Brad King <brad.king@kitware.com>
Date: Wed, 18 Nov 2020 07:46:23 -0500
Subject: [PATCH] CMake 3.19.0
```

```
[Add azure-pipelines and e2e tests to vcpkg-tool.](https://github.com/microsoft/vcpkg-tool/commit/7881dfc9e2ce719907c0b5391df6b5706e7fda13)

@[BillyONeal](https://github.com/microsoft/vcpkg-tool/commits?author=BillyONeal)
BillyONeal committed on Feb 3, 2021
```

Seems like `BillyONeal committed on Feb 3, 2021` predates `Date: Wed, 18 Nov 2020 07:46:23 -0500` in your frame of reference. How fast are you moving that time dilatation is that big? (I am just evil at this point but lets call it that both were created around the same time and the benefits of presets weren't know yet)

Reply:
> Seems like BillyONeal committed on Feb 3, 2021 predates Date: Wed, 18 Nov 2020 07:46:23 -0500 in your frame of reference.

When it was committed in CMake's tree vs. when it was available for us to use are very different things, but fine, drop "long" from my statement.